### ISSUE TITLE
Less prominent status indicator

### ISSUE THREAD

**Issue by davicostalf** [vtex/shoreline#1814]
### Summary

Currently we have tags, but they only work well in less information dense tables, right next to the menu. Should we have a more discrete way to indicate status? Are there studies in other scenarios? When should a regular Tag be used?

![Publisher - Reports](https://github.com/user-attachments/assets/7f025979-dfd4-480b-a322-11f8c2b925d3)
![Publisher - Reports-1](https://github.com/user-attachments/assets/83d5f2bb-1ddd-42ca-8de3-7c5a2cd7c263)

**Comment by beatrizmilhomem** [comment]
I've seen lots of studies with a solution similar to your second one when it comes to representing status, I think that it makes sense to have this pattern at this point.

Just as a first suggestion regarding the application, tags could be more category-related.

**Comment by beatrizmilhomem** [comment]
What other types of statuses have been used besides active/inactive?

**Comment by elleenvs-d** [comment]
Severity could be another type? If so, we had this type to indicate risk degree for findings. 
<img width="674" alt="Captura de Tela 2024-08-16 às 14 14 31" src="https://github.com/user-attachments/assets/a614784d-79dc-4ba2-8543-e275c144a822">

**Comment by lafray** [comment]
I wanted to mention that we have had apps in the Marketplace that needed a component like this (at the time it was still admin-ui and shorelina didn't exist). And at the time we ended up going for a custom component as well.

I wanted to mention it because I think it would bring a lot of value to have this here in the lib.

**Comment by beatrizmilhomem** [comment]
Next step: make a proposal for the component.

**Comment by luarakerlen** [comment]
We created a StatusIndicator locally to use it at admin ads reports: https://github.com/vtex/admin-ads/pull/125

![image](https://github.com/user-attachments/assets/0eec1281-de76-42e9-ac35-bc570cd7ad2c)

The possible status, in this case, are: `Processing`, `Error` and `Available`.

@davicostalf asked me to add this here as an architectural proposal.

**Comment by lafray** [comment]
Você estão vendo esse componente sendo com status mais fechados (como o que tem na [doc](https://vercel.com/geist/status-dot) que o @matheusps trouxe e do que a @luarakerlen trouxe do `StatusIndicator`) ou algo mais aberto como é a tag do shoreline onde só é definido colors, varianets and size?

Tenho uma preocupação quanto e ter status fechados, pois acho que a vtex tem casos muito diversos onde pode ser dificil de criar um conjunto de status que resoonda bem a maioria dos casos.

**Comment by beatrizmilhomem** [comment]
> Tenho uma preocupação quanto e ter status fechados, pois acho que a vtex tem casos muito diversos onde pode ser dificil de criar um conjunto de status que resoonda bem a maioria dos casos.

Concordo, iria pra um caminho mais aberto mesmo no componente e documentaria exemplos de aplicação de possíveis status.

**Comment by beatrizmilhomem** [comment]
### Proposta

- **Componente e testes:** [Figma](https://www.figma.com/design/oz3BlwaVvXEVOQTFa4ZdNj/Status-Indicator?node-id=3-106&t=KQdNvOnV8ese5o26-1)
- **Observações:** Pra diferenciação entre as cores ficar mais clara e as cores semânticas terem um peso semelhante, usei tons entre 6 e 9. Também estou propondo usar `text-emphasis` ao invés de `text-body`, porque é o que usamos em `Tag` também e faria sentido ter esses componentes de "classificação" seguindo a mesma lógica.
- **Descrição do StatusIndicator:** StatusIndicators differentiate the status of items through a colored dot and a read-only text value. Status can be related to availability, severity, or activity.
- **Descrição da Tag** (para diferenciar as aplicações, deve ter uma atualização na sua descrição): Tags differentiate the category of items through a read-only text value in a colored bg. They can be primary or secondary, given their relevance to the main job.

**Comment by davicostalf** [comment]
@beatrizmilhomem Como vamos indicar o status de itens no page header? Continuamos usando uma Tag nesses cenários?

**Comment by beatrizmilhomem** [comment]
> Como vamos indicar o status de itens no page header? Continuamos usando uma Tag nesses cenários?

Fiz uns [testes](https://www.figma.com/design/oz3BlwaVvXEVOQTFa4ZdNj/Status-Indicator?node-id=3-106) aqui. Pensei em usar o [sem bg no meio de conteúdo e com bg em header](https://www.figma.com/design/oz3BlwaVvXEVOQTFa4ZdNj/Status-Indicator?node-id=23-62377&t=m4AojfQlGFfia5ex-4) (quando não tem bg no header [ficou meio solto](https://www.figma.com/design/oz3BlwaVvXEVOQTFa4ZdNj/Status-Indicator?node-id=23-49411&t=m4AojfQlGFfia5ex-4)). O que acha?

> Quais são as regras de aplicação do text-emphasis? Será que tem a ver com um fundo "colorido" precisar de mais contraste?

Hoje temos isso na doc: "Token with a heavier weight than `$text-body`. Use sparingly in content that is non-interactive or descriptive (Alert message, column name in a Table) or in actions that are less relevant than Buttons (Filters and Menu items). ", tá até faltando citar Tag. Mas tem mais a ver com peso do que com fundo.

**Comment by lafray** [comment]
Uma coisa não fica muito clara para mim. Quando se usaria o StatusIndicator com BG no lugar de uma tag? A versão com BG deixa essa distinção de uso menos clara para mim.

**Comment by davicostalf** [comment]
> Fiz uns testes aqui. Pensei em usar o sem bg no meio de conteúdo e com bg em header (quando não tem bg no header ficou meio solto). O que acha?
Ficou muito solto sem o BG e achei meio solto com o BG também. Talvez abaixo do page title?

> Quando se usaria o StatusIndicator com BG no lugar de uma tag? A versão com BG deixa essa distinção de uso menos clara para mim.
Concordo com Lais. Quando se adiciona o BG a distinção fica mais fraca. Talvez aceitar o status indicator como uma variante da Tag?

**Comment by malu-viana** [comment]
Trazendo mais dois casos de uso nos cenários de marketplace:

<img width="1440" alt="OfferQuality" src="https://github.com/user-attachments/assets/f8a6c005-d685-4527-afbf-35f8ef5de80f"> <img width="1440" alt="Offer status" src="https://github.com/user-attachments/assets/5713e7b5-16e9-448d-98c0-d1554098f442">

Esse uso no início da linha, assim como no catálogo, economiza espaço e não conflita com o uso de outra coluna com tag. Pode ser uma variação a ser considerada, Icon only.

> Ficou muito solto sem o BG e achei meio solto com o BG também. Talvez abaixo do page title?
Fiquei pensando que antes do PageTitle, da mesma forma que no exemplo de uso Icon only na tabela

**Comment by beatrizmilhomem** [comment]
> Fiquei pensando que antes do PageTitle, da mesma forma que no exemplo de uso Icon only na tabela

Fiz uns testes aqui e acho que fez sentido.
Pro caso de IconOnly, adicionaria na documentação de boas práticas que sempre deve ter tooltip no hover. Não incluí no componente porque a posição da tooltip pode variar.
https://www.figma.com/design/oz3BlwaVvXEVOQTFa4ZdNj/Status-Indicator?node-id=53-13132&t=QZ879Pkd1y9tUz06-4

**Comment by malu-viana** [comment]
@beatrizmilhomem gosto bastante!
